modules/candidate_parameters/jsx/monitoring.js

```diff
@@ -0,0 +1,449 @@
+/* exported RMonitoringStatus */
+
+var MonitoringStatus = React.createClass(
+  {
+    getInitialState: function() {
+      return {
+        flaggedOptions: {
+          yes: "Yes",
+          no: "No"
+        },
+        monitoredOptions: {
+          Y: "Yes",
+          N: "No"
+        },
+        Data: [],
+        formData: {},
+        updateResult: null,
+        errorMessage: null,
+        isLoaded: false,
+        loadedData: 0
+      };
+    },
+    componentDidMount: function() {
+      var that = this;
+      $.ajax(
+        this.props.dataURL,
+        {
+          dataType: 'json',
+          xhr: function() {
+            var xhr = new window.XMLHttpRequest();
+            xhr.addEventListener(
+              "progress",
+              function(evt) {
+                that.setState(
+                  {
+                    loadedData: evt.loaded
+                  }
+                );
+              }
+            );
+            return xhr;
+          },
+          success: function(data) {
+            var formData = {};
+            var visits = data.visits;
+            for (var visit in visits) {
+              if (visits.hasOwnProperty(visit)) {
+                var monitored = visit + "_monitored";
+                var dateMonitored = visit + "_date";
+                var dateMonitored2 = visit + "_date2";
+                var monitorID = visit + "_monitor";
+                formData[monitored] = data.monitored[visit];
+                formData[visit] = data.flagged[visit];
+                formData[dateMonitored] = data.dateMonitored[visit];
+                formData[dateMonitored2] = data.dateMonitored[visit];
+                formData[monitorID] = data.monitorID[visit];
+              }
+            }
+
+            that.setState(
+              {
+                Data: data,
+                formData: formData,
+                isLoaded: true
+              }
+            );
+          },
+          error: function(data, errorCode, errorMsg) {
+            that.setState(
+              {
+                error: 'An error occurred when loading the form!'
+              }
+            );
+          }
+        }
+      );
+    },
+    setFormData: function(formElement, value) {
+      var formData = this.state.formData;
```

Comment:
Deep copy of state before mutating
`const formData = JSON.parse(JSON.stringify(this.state.formData))`